### ISSUE TITLE
Linuxkit support

### ISSUE THREAD

**Issue by hyperized** [linuxkit/linuxkit#3437]
A bit of a meta issue here, but:

How is this project supported?
The only thing I can find is that currently two people are assigned to the project.
(Maybe: Can you document this?)

We're really liking the way Linuxkit is working and we're looking to use Linuxkit as a building block for our definition driven infrastructure going forward.

However I do notice there's a bit of a drop off in activity since the end of 2018 and it seems that keeping this project up to date is not getting a unified effort.

Ideally I'd like to be able to run Kubernetes on this, but currently there's a lot of hoops to jump through to get 1.16 properly supported.

There are a few PR's open that sort of overlap, but they have been quite silent for a bit:

- https://github.com/linuxkit/linuxkit/pull/3426
- https://github.com/linuxkit/linuxkit/pull/3388
- https://github.com/linuxkit/linuxkit/pull/3423

Is there any way we can help with this? How?
And going forward, will this project receive continuous support from Docker?

**Comment by justincormack** [comment]
Hi. The current situation is that there are three active maintainers, but none of us are working full time on the project at present. Avi is available for consulting work. Docker are using the project actively, as are many other organizations, but mostly people are using as is, using their own containers but not making significant core modifications that we are aware of. There are a bunch of projects that need a significant investment of time to do, which would make things way more usable, in particular reworking CI, integrating the filesystem work, and integrating containerd in the build process.

Those three PRs are things that just need finishing off, and someone can contribute them. The current system for building images is annoying though, and currently only maintainers can do it, part of the CI rework would be building containers in CI so they can be tested by anyone and then built in CI, rather than the current manual process, so that would make contribution way easier.

We are happy to add new maintainers if people want to contribute.

**Comment by deitch** [comment]
I can only second what Justin said. There are some improvements that we as a team would like to get done, and then there are community/customer-driven improvements we would like to see. We would be happy both to have more contributors, and if it is a business-specific need, we can look at a consulting arrangement as well.

**Comment by hyperized** [comment]
Thank you for the reponses, it's given us a clearer picture as to what to expect in terms of support.

We're currently not interested in consultancy, but we'll know where to look for when we do :).

Might it be an idea to map out these goals as milestones (under Projects) and then link the issues to them so it's easier for a non-maintainer like myself to see what needs attention?

I would definately be interested in contributing and I think a decently working CI/CD setup would be a requirement. Would the new Github Actions be something for this project?

**Comment by justincormack** [comment]
Yes, I think actions may be a very good fit, especially now they support bring your own server which makes supporting arm64 etc easier. We want to be able to build and test all the containers built in a PR, then on merge push them to the main repo.
